openlabcmd/openlabcmd/plugins/nodepool/orphanresource.py

```diff
@@ -0,0 +1,120 @@
+import copy
+import subprocess
+
+from openlabcmd.plugins.base import Plugin
+
+SERVER_WHITE_LIST = [
+    '%s-openlab-zuul', '%s-openlab-nodepool', '%s-openlab-zookeeper']
+
+
+class OrphanResourcePlugin(Plugin):
+
+    # This Plugin is only for checking, as it's a dangerous action for each.
+    ptype = 'nodepool'
+    name = 'OrphanResource'
+
+    def check(self):
+        self.failed = False
+        if self.reasons:
+            self.reasons.pop()
+        server_white_list = [elemnt % self.cloud
+                             for elemnt in copy.deepcopy(SERVER_WHITE_LIST)]
+
+        # For floatingip, we will check whether a fip didn't associated with a
+        # neutron port.
+        get_fip_cmd = ("openstack --os-cloud %s floating ip list "
+                       "-c 'ID' -c 'Floating IP Address' -c 'Port' | "
+                       "egrep '[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-"
```

Comment:
maybe such as commands should be more simple ?
`openstack floating ip list -c 'Floating IP Address' -c 'Fixed IP Address' -f value |grep None| awk '{ print $1 }'`

Reply:
Yeah, it works. Thanks.